crates/containerd-shim-wasm/src/sandbox/cli.rs

```diff
@@ -36,6 +40,47 @@ macro_rules! revision {
     };
 }
 
+/// Main entry point for the shim with OpenTelemetry tracing.
+///
+/// It parses the `OTEL_EXPORTER_OTLP_ENDPOINT` environment variable to determine
+/// if the shim should be started with OpenTelemetry tracing.
+///
+/// If the environment variable is not set, the shim will be started without tracing.
+/// If the environment variable is empty, the shim will be started without tracing.
+#[cfg(feature = "opentelemetry")]
+pub fn shim_main_with_otel<'a, I>(
```

Comment:
**thought:** I think I would prefer another generic argument (with a default value) to `shim_main`, which is a trait for a tracing provider. Then the otel tracer is just one impl of that tracer. The default value could be a no-op provider. Or maybe make the tracing trait part of the shim's `config` argument. I think that would also help with mocking in tests. But it's not a blocker, I am happy for this to go in as is as there's a lot of value in this PR, we can tweak things in a follow up PR.

Reply:
Oh I think the main reason I have a seperate function for `shim_main_with_otel` is to be able to trace `shim_main` function which is doing a lot of work. Initializing tracer provider **in the function** does not export trace of that function iiuc.

Reply:
Could we rename
`pub fn shim_main_with_otel<'a, I>(...)` -> `pub fn shim_main<'a, I>(...)`
`pub fn shim_main(...)` -> `fn shim_main_inner(...)` (not pub)
Then we only expose one of the methods to consumers.
Happy to discuss this on slack :-)

Reply:
Done